spec/system/editing_a_reservation_spec.rb

```diff
@@ -91,4 +91,96 @@
       expect(page).to have_content "cannot be shortened once the reservation has started"
     end
   end
+
+  describe "with a daily booking instrument" do
+    let(:today) { Time.current.beginning_of_day }
+    let(:unavailable_beginning_of_day) { today + 3.days }
+    let(:instrument) do
+      create(
+        :setup_instrument,
+        :always_available,
+        :daily_booking,
+      )
+    end
+    let!(:reservation) do
+      create(
+        :purchased_reservation,
+        user:,
+        reserve_start_at: today + 10.days,
+        reserve_end_at: today + 11.days,
+        product: instrument,
+      )
+    end
+    before do
+      create(
+        :purchased_reservation,
+        user: create(:user),
+        reserve_start_at: today + 1.day,
+        reserve_end_at: today + 2.days,
+        product: instrument,
+      )
+    end
+
+    shared_examples "move a reservation to end of day" do |error_key|
+      it "ensure instrument is not available on the unavailable day" do
+        reserve_end_at = unavailable_beginning_of_day + 1.minute
+        invalid_reservation = build(
+          :purchased_reservation,
+          user:,
+          reserve_start_at: reserve_end_at - 1.day,
+          reserve_end_at:,
+          product: instrument
+        )
+
+        expect(invalid_reservation).to_not be_valid
+        expect(invalid_reservation.errors.map(&:type)).to include(error_key)
+      end
```

Comment:
Sorry, is this case when a reservation ends on an unavailable day? Isn't that case supposed to be valid or are we leaving that for the next ticket?

Reply:
this is just checking that the unavailable day is actually unavailable due the reason specified on `error_key`